software/webapp/src/main/java/org/apache/brooklyn/entity/webapp/JavaWebAppSshDriver.java

```diff
@@ -93,11 +93,11 @@ protected String inferRootUrl() {
         if (isProtocolEnabled("https")) {
             Integer port = getHttpsPort();
             checkNotNull(port, "HTTPS_PORT sensors not set; is an acceptable port available?");
-            return String.format("https://%s:%s/", getSubnetHostname(), port);
+            return String.format("https://%s:%s/", getHostname(), port);
```

Comment:
@neykov @aledsage I think that `BrooklynAccessUtils` is actually the right choice to get the host and port for the `main.uri` URL which will therefore work with Clocker. Note that the current `getSubnetHostname()` call will return the SDN provided IP address, so with e.g. Calico defaults in Clocker you would get 50.0.0.2 as the address, and that is only accessible from other containers, not from Brookln. And `getHostname()` will return the Docker host IP address, so the port will then be wrong, due to forwarding.